test/instruments_previous_test.go

```diff
@@ -112,5 +149,21 @@ func createTokenInstrumentPrevious(t *testing.T, token *tokens.CardTokenResponse
 	response, err := PreviousApi().Instruments.Create(request)
 	assert.Nil(t, err)
 	assert.NotNil(t, response)
+	assert.Equal(t, instruments.Card, response.Type)
+	assert.NotEmpty(t, response.Id)
+	assert.NotEmpty(t, response.Fingerprint)
+	assert.NotEmpty(t, response.ExpiryMonth)
+	assert.NotEmpty(t, response.ExpiryYear)
+	assert.NotEmpty(t, response.Scheme)
+	assert.NotEmpty(t, response.Last4)
+	assert.NotEmpty(t, response.Bin)
+	assert.NotEmpty(t, response.CardType)
+	assert.NotEmpty(t, response.CardCategory)
+	assert.NotEmpty(t, response.ProductId)
+	assert.NotEmpty(t, response.ProductType)
+	assert.NotEmpty(t, response.Customer)
+	assert.NotEmpty(t, response.Customer.Id)
+	assert.NotEmpty(t, response.Customer.Name)
+	assert.NotEmpty(t, response.Customer.Email)
```

Comment:
This should all be asserted in the TestCreateInstrument function. Here we should only validate if there was an error in the response and in that case do a t.Fail("with some error message"). You can check the `createWorkflow(t *testing.T) *common.IdResponse` function

Reply:
👍